### PR TITLE
docs(checkbox): add yard

### PR DESCRIPTION
**Pull request by gergelyke**

### CODE REVIEW

File: documentation-site/components/yard/config/checkbox.ts [hunk 1]
```diff
@@ -20,8 +21,19 @@ export default {
...
+    },
+    checkmarkType: {
+      value: undefined,
+      options: STYLE_TYPE,
+      type: PropTypes.Enum,
+      description: 'Renders component in a toggle state.',
+    },
     onChange: {
-      value: 'e => setChecked(e.target.checked)',
+      value: 'e => setChecked(!e.target.checked)',
```
Comment (tajo) [inline documentation-site/components/yard/config/checkbox.ts]: hm, `!` should not be here?

Comment (gergelyke) [reply documentation-site/components/yard/config/checkbox.ts]: yeah, that's correct

File: documentation-site/components/yard/config/checkbox.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import {Checkbox, STYLE_TYPE} from 'baseui/checkbox';
+import {PropTypes} from '../const';
+
+export default {
+  scopeConfig: {
+    Checkbox,
+  },
+  themeConfig: [],
```
Comment (nadiia) [inline documentation-site/components/yard/config/checkbox.ts]: Do we need to fill out the `themeConfig`?

File: documentation-site/components/yard/config/checkbox.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          into: 'checked',
+        },
+      },
+    },
+    overrides: {
+      value: undefined,
+      type: PropTypes.Overrides,
+      description: 'Lets you customize all aspects of the component.',
+      meta: {
+        names: ['Root'],
```
Comment (nadiia) [inline documentation-site/components/yard/config/checkbox.ts]: Not sure if the PR is still in progress but there are more overrides to it.

Comment (gergelyke) [reply documentation-site/components/yard/config/checkbox.ts]: yes, it is a work in progress

**Comment by gergelyke** [comment]
@nadiia @tajo the PR should be ready for a review

File: documentation-site/components/yard/code-generator.ts [hunk 4]
```diff
@@ -118,7 +119,8 @@ export const getEnumsToImport = (props: {[key: string]: TProp}) => {
   const enums: string[] = [];
   Object.keys(props).forEach(name => {
     if (props[name].type === PropTypes.Enum && props[name].value) {
-      enums.push(name.toUpperCase());
+      console.log(props[name].enumName, name);
```
Comment (tajo) [inline documentation-site/components/yard/code-generator.ts]:
```suggestion
```

File: documentation-site/components/yard/config/checkbox.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+      description: 'A ref to access an input element.',
+      hidden: true,
+    },
+    autoFocus: {
+      value: false,
+      type: PropTypes.Boolean,
+      description: 'If true the component will be focused on the first mount.',
+      hidden: true,
+    },
+    type: {
```
Comment (tajo) [inline documentation-site/components/yard/config/checkbox.ts]: hm, what is this prop for? should it always be "checkbox"?

at least I would set `hidden: true`, doesn't seem important

Comment (gergelyke) [reply documentation-site/components/yard/config/checkbox.ts]: yeah, the default value for this is checkbox - I'll just remove this from yard

**Pull request merged**